api/renterd/api.go

```diff
@@ -0,0 +1,211 @@
+	}
+	max := -1
+	if v := req.FormValue("max"); v != "" {
+		t, err := strconv.Atoi(v)
+		if err != nil {
+			http.Error(w, "invalid max value: "+err.Error(), http.StatusBadRequest)
+			return
+		}
+		max = t
+	}
+	api.WriteJSON(w, s.w.Transactions(since, max))
+}
+
+func (s *server) walletSignHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var wsr WalletSignRequest
+	if err := json.NewDecoder(req.Body).Decode(&wsr); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	txn, toSign := wsr.Transaction, wsr.ToSign
+	if len(toSign) == 0 {
+		// lazy mode: add standard sigs for every input we own
+		for _, input := range txn.SiacoinInputs {
+			toSign = append(toSign, input.Parent.ID)
+		}
+	}
+
+	vc, err := s.cm.TipContext()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if err := s.w.SignTransaction(vc, &txn, nil); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	api.WriteJSON(w, WalletTransactionResponse{txn})
+}
+
+func (s *server) txpoolBroadcastHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var tbr TxpoolBroadcastRequest
+	if err := json.NewDecoder(req.Body).Decode(&tbr); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	for _, txn := range tbr.DependsOn {
+		if err := s.tp.AddTransaction(txn); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+	}
+	if err := s.tp.AddTransaction(tbr.Transaction); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	s.s.BroadcastTransaction(tbr.Transaction, tbr.DependsOn)
+}
+
+func (s *server) txpoolTransactionsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	api.WriteJSON(w, s.tp.Transactions())
+}
+
+func (s *server) syncerPeersHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	ps := s.s.Peers()
+	sps := make([]SyncerPeerResponse, len(ps))
+	for i, peer := range ps {
+		sps[i] = SyncerPeerResponse{
+			NetAddress: peer,
+		}
+	}
+	api.WriteJSON(w, sps)
+}
+
+func (s *server) syncerConnectHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	var scr SyncerConnectRequest
+	if err := json.NewDecoder(req.Body).Decode(&scr); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err := s.s.Connect(scr.NetAddress); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+}
+
+// NewHandler returns a new http handler for the renter API endpoints.
+func NewHandler(cm ChainManager, s Syncer, w Wallet, tp TransactionPool) http.Handler {
+	const password = "testing"
+
+	srv := server{
+		cm: cm,
+		s:  s,
+		w:  w,
+		tp: tp,
+	}
+	mux := httprouter.New()
+
+	mux.GET("/api/wallet/balance", api.AuthMiddleware(srv.walletBalanceHandler, password))
+	mux.GET("/api/wallet/address", api.AuthMiddleware(srv.walletAddressHandler, password))
```

Comment:
Since the web UI will be the entry point for everyday users, I think prefixing the API and leaving the web assets on `/` is better. I believe we could use `httprouter.NotFoundHandler` to server the web assets and API together unprefixed, but that seems hacky. I lean towards prefixed for the clear separation between UI and data routes.

Reply:
ah, forgot about that.

>  I believe we could use `httprouter.NotFoundHandler`

yeah, @alexfreska had the same thought. 😅 I agree with your take.